### DISCUSSION TITLE
Turtle API Question

### DISCUSSION THREAD

**Discussion by SirEndii** [cc-tweaked/CC-Tweaked#724] [Q&A] [answered]
Hey,
I wanted to create some Turtles for my Mod. I use `ComputerCraftAPI#registerTurtleUpgrade` to register a turtle and the Interface `ITurtleUpgrade` to create my turtle. I use registerTurtleUpgrade in an `RegistryEvent.Register<Item>` Event, like CC:Tweaked does it([Registry.java](https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/shared/Registry.java#L238)). 

But nothing happens, no turtle ingame.
My Turtle class: [Class](https://github.com/Seniorendi/AdvancedPeripherals/blob/master/src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/turtles/TurtleChatBox.java) *I use the models from CC just for testing*

My Registry class: [Class](https://github.com/Seniorendi/AdvancedPeripherals/blob/master/src/main/java/de/srendi/advancedperipherals/common/setup/Turtles.java)

**Comment by SquidDev** [comment, marked as answer]
`public void registerItems` needs to be static if it's registered with `@Mod.EventBusSubscriber`. You shouldn't be touching `TurtleUpgrades` (and neither should CC's registry really) - that's an internal class.

**Reply by SirEndii** [reply]
registerItems is static, the commit is not on github. I also do not use TurtleUpgrades anymore, I will push the latest commit on github.

**Reply by SirEndii** [reply]
The registering kinda works, when i try to register the turtle twice, the game crashes with [this](https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/shared/TurtleUpgrades.java#L60) Exception.
I pushed the commit on github, see [here](https://github.com/Seniorendi/AdvancedPeripherals/commit/cb132187defde90afc1eafed2ae7f1a067630411).

**Reply by SquidDev** [reply]
> when i try to register the turtle twice, the game crashes 

I'm a little confused - where (and why) are you registering things twice?

**Reply by SirEndii** [reply]
> I'm a little confused - where (and why) are you registering things twice?

I just tried it if my game crashes when I try to register my turtle twice. 
CC registers my turtle, but my turtle is not ingame.

**Reply by SirEndii** [reply]
My turtle works now. I can place it, craft it, ... . But, is it possible to queue events for turtles, to use os.pullEvent?

**Reply by Lupus590** [reply]
Via this API I would assume: https://github.com/SquidDev-CC/CC-Tweaked/blob/1f70ed69854de12356681f4036458894a5848c0d/src/main/java/dan200/computercraft/api/peripheral/IComputerAccess.java#L153
